### PR TITLE
210 endpoint to unlike an article

### PR DESCRIPTION
**Pull request by itzgeebee**
#### Pre-Submission Checklist

-   [x] Your pull request targets the `main` branch.
-   [x] Branch name is descriptive and follows the guidelines [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices).
-   [x] All tests pass . Use `git commit --amend` to amend any fixes.

#### Checklist:

-   [x] All tests are passing.
-   [x] Tested changes locally.
-   [x] Improved documentation or readme where necessary.
-   [x] Your commit messages format follows our commit message guideline [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices#commits).

#### What does this PR do?

This PR adds a feature to delete article likes

#### How should this be manually tested?

<!-- Add specific steps on how to test this manually
1. npm run test
2. npm run dev

### CODE REVIEW

File: src/services/posts/delete-article-like.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const {expect} = require('chai')
+const db = require("../../db")
+const {fixtures} = require('../../../test/utils')
+const deleteArticleLike = require("./delete-article-like")
```
Comment (jocrah) [inline src/services/posts/delete-article-like.test.js:4]: @itzgeebee let's use unlikePost for our purposes. same for the file name

**Comment by jocrah** [comment]
@itzgeebee, also this is getting too easy 😄  so for this PR, we will take it another step and start adding api testing.

We'll be using the already installed supertest package. Kindly do check out resources in relation to that. 

this is a sample resource:  https://medium.com/hackernoon/api-testing-using-supertest-1f830ce838f1

Take a look at the intended structure from the Technical Considerations here:  https://github.com/tbf-challenge/teamwork-be/issues/198

**Comment by itzgeebee** [comment]
> @itzgeebee, also this is getting too easy smile so for this PR, we will take it another step and start adding api testing.
> 
> We'll be using the already installed supertest package. Kindly do check out resources in relation to that.
> 
> this is a sample resource: https://medium.com/hackernoon/api-testing-using-supertest-1f830ce838f1
> 
> Take a look at the intended structure from the Technical Considerations here: #198

Haha. Okay boss

**Comment by jocrah** [comment]
The test file will have one main `describe` block and then two `describe` blocks one called `Failure` and the other called `Success`

the Failure  test cases:
1. should return 400 if postId is not a number (it should check for the right error message as well)
2.  should return 400 if userId is not a number
3. should return 401 if request is not authenticated

the Success  test cases:

1. should return 200
2. it should return right response


@itzgeebee  cc @beblicarl

**Comment by jocrah** [comment]
@itzgeebee, I've stressed @beblicarl  for us to get a sample api endpoint test in production : https://github.com/tbf-challenge/teamwork-be/pull/219

That can be a blueprint for what you do in this one

Kindly feel free to ask any questions about the parts of it which are not clear, thanks

**Comment by itzgeebee** [comment]
> @itzgeebee, also this is getting too easy smile so for this PR, we will take it another step and start adding api testing.
> 
> We'll be using the already installed supertest package. Kindly do check out resources in relation to that.
> 
> this is a sample resource: https://medium.com/hackernoon/api-testing-using-supertest-1f830ce838f1
> 
> Take a look at the intended structure from the Technical Considerations here: #198


> @itzgeebee, I've stressed @beblicarl for us to get a sample api endpoint test in production : https://github.com/tbf-challenge/teamwork-be/blob/main/src/routes/gifs/unlike-gif.e2e.test.js
> 
> That can be a blueprint for what we do here and moving forward
> 
> Kindly feel free to ask any questions about the parts of it which are not clear, thanks

Alright. Thank you

**Comment by itzgeebee** [comment]
All done @jocrah

**Comment by beblicarl** [comment]
> All done @jocrah

that was fast lol

**Comment by itzgeebee** [comment]
> > All done @jocrah
> 
> that was fast lol

@beblicarl You already did all the work

**Comment by jocrah** [comment]
> All done @jocrah

okay, taking a look

File: src/schema.js [hunk 2]
```diff
@@ -52,8 +52,6 @@ const signinSchema = Joi.object({
 
 const createArticleSchema = Joi.object({
 	body:{
-		userId: Joi.number()
```
Comment (jocrah) [inline src/schema.js:55]: @itzgeebee, this intentional?

File: src/services/posts/unlike-post.test.js [hunk 3]
```diff
@@ -5,32 +5,53 @@ const unlikePost = require("./unlike-post")
 const {fixtures} = require('../../../test/utils')
 
 
-describe('UNLIKE a gif', () => {
+describe('UNLIKE a post', () => {
```
Comment (jocrah) [inline src/services/posts/unlike-post.test.js:8]: this is cool, but let's deal with the article and gif tests here in this same file as as inner `describe` blocks. Kindly check out the format here: https://github.com/tbf-challenge/teamwork-be/blob/main/src/services/posts/like-post.test.js

File: src/schema.js (hunk 2 above)
Comment (itzgeebee) [reply src/schema.js:55]: @jocrah Nah, I probably forgot to add it back after testing something

File: src/services/posts/unlike-post.test.js (hunk 3 above)
Comment (itzgeebee) [reply src/services/posts/unlike-post.test.js:8]: @jocrah Done

Comment (jocrah) [reply src/services/posts/unlike-post.test.js:8]: okay, let's kindly take out the `delete-article.test.js` then

File: src/schema.js [hunk 4]
```diff
@@ -52,8 +52,7 @@ const signinSchema = Joi.object({
...
-			.required(),
+		userId: Joi.number(),
```
Comment (jocrah) [inline src/schema.js:55]: @itzgeebee this is slightly different from what we had

to get the original file before the change you could simply run

```
git checkout origin/main src/schema.js
```
and commit and push it

Comment (itzgeebee) [reply src/schema.js]: fixed it. Thanks @jocrah

File: src/services/posts/unlike-post.test.js (hunk 3 above)
Comment (itzgeebee) [reply src/services/posts/unlike-post.test.js:8]: @jocrah Done

**Review by jocrah — APPROVED** [review]

**Pull request merged**